app/components/branch-row.js

```diff
@@ -19,14 +20,20 @@ export default Component.extend({
   isTriggering: false,
   hasTriggered: false,
 
-  commitUrl: computed('branch.repository.{slug,vcsType}', 'branch.last_build.commit.sha', function () {
-    const [owner, repo] = this.get('branch.repository.slug').split('/');
-    const vcsType = this.get('branch.repository.vcs_type');
+  commitUrl: computed('branch.repository.id', 'branch.last_build.commit.sha', function () {
+    const repository = this.store.peekRecord('repo', this.get('branch.repository.id'));
+    const owner = repository.owner.login;
+    const repo = repository.name;
+    const vcsType = repository.vcsType;
```

Comment:
I have some concerns about this change. Generally, computed properties should operate only with data that's listed in the dependent keys. Here, `store` is another variable that's not tracked by the computed property. If `store` will change, this property won't recompute, thus getting out of sync and putting the app to invalid state. 

I can see that `branch` has a relation for [`repo`](https://github.com/travis-ci/travis-web/blob/master/app/models/branch.js#L11). Wouldn't it work without the need to `peekRecord` directly from store? Something like:
```javascript
const { owner, name, vcsType } = this.branch.repo;
```
To be honest, I'm not sure what's the `repository` that the code here is referring to 🤔